src/components/ReportActionItem/MoneyRequestAction.tsx

```diff
@@ -87,31 +70,32 @@ function MoneyRequestAction({
     reportID,
     isMostRecentIOUReportAction,
     contextMenuAnchor,
-    checkIfContextMenuActive,
+    checkIfContextMenuActive = () => {},
     chatReport,
     iouReport,
     reportActions,
-    isHovered,
-    network,
+    isHovered = false,
     style,
-    isWhisper,
-}) {
+    isWhisper = false,
+}: MoneyRequestActionProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
-    const isSplitBillAction = lodashGet(action, 'originalMessage.type', '') === CONST.IOU.REPORT_ACTION_TYPE.SPLIT;
+    const {isOffline} = useNetwork();
+
+    const isSplitBillAction = action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && action.originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.SPLIT;
 
     const onMoneyRequestPreviewPressed = () => {
         if (isSplitBillAction) {
-            const reportActionID = lodashGet(action, 'reportActionID', '0');
+            const reportActionID = action.reportActionID ?? '0';
             Navigation.navigate(ROUTES.SPLIT_BILL_DETAILS.getRoute(chatReportID, reportActionID));
             return;
         }
 
         // If the childReportID is not present, we need to create a new thread
-        const childReportID = lodashGet(action, 'childReportID', 0);
+        const childReportID = action?.childReportID ?? '0';
```

Comment:
@VickyStash heads up! that all IDs must be numbers, this created this bug https://github.com/Expensify/App/issues/35904. :)

Reply:
@youssef-lr reportIDs are expected to be string, otherwise they can overflow, right?

Reply:
Yeah sorry my bad, they are all numbers in the backend, but in JS `reportIDs` & `reportActionIDs` are strings. In this specific example it doesn't make sense to use `'0'`, because the condition below this line evaluates to `false` when it should evaluate to `true` `if (!childReportID) {`

Reply:
@youssef-lr you are right, my bad!

Reply:
Commented in your PR @youssef-lr , I don't think we should default to 0 either

Reply:
Yeah that sounds good, updated.